CONTRIBUTING.md

```diff
@@ -0,0 +1,94 @@
+
+If you discover a problem with a REST API document in this repo, feel free to [open an issue](https://github.com/Azure/azure-rest-api-specs/issues/new). But please do not report issues with service behavior / functionality in this repo.
+
+When opening an issue, please clearly describe the problem including which REST API definition is in error and what elements of the definition are incorrect or missing.
+
+## Avoid Breaking Changes
+
+The first rule for Azure REST APIs is to avoid breaking changes in GA'd APIs. The [Azure Breaking Changes Policy](aka.ms/AzBreakingChangesPolicy) is a formal description of what changes are considered breaking changes. This rule trumps adherence to Design Guidelines, Coding Style, etc.
+
+## Design Guidelines
+
+Azure REST APIs should conform to the [Azure REST API Guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md).
+
+There is a [YouTube video series](https://www.youtube.com/watch?v=9Ng00IlBCtw) by Jeffrey Richter that describes the guidelines in detail.
+
+Another resource is the [Considerations for Service Design](https://github.com/microsoft/api-guidelines/blob/vNext/azure/ConsiderationsForServiceDesign.md), which is an introduction to REST API design mainly for services that are just getting started.
+
+### Exceptions for consistency within a service
+
+There are situations where a service has GA'd their API with design patterns that do not follow our guidelines and it would be a breaking change to adopt the API design in the guidelines.
+Because the first rule is to avoid breaking changes and because we want APIs to be consistent within a service, these design patterns are considered the standard for that service and should be followed in all subsequent (non-breaking) versions of that service's REST API.
+
+## Coding Style
+
+REST APIs for Azure should be defined using the [OpenAPI v2](https://github.com/OAI/OpenAPI-Specification/blob/main/versions/2.0.md) format.
+
+The REST API should also adhere to the conventions documented in the [Azure API Style Guide](https://github.com/Azure/azure-api-style-guide/blob/main/openapi-style-guide.md).
+
+You can use the [Spectral linter](https://meta.stoplight.io/docs/spectral/674b27b261c3c-overview) with our [Spectral ruleset](https://github.com/Azure/azure-api-style-guide#how-to-use-the-spectral-ruleset) to check conformance with OpenAPI v2 and the Style Guide.
+
+## Directory Structure
+
+See the [README.md][./README.md] for a description of the directory structure to use when adding files to this repo.
+
+## Pull Requests
+
+If you want to contribute to the repository, follow these steps:  
+  1. Fork the repository and create a new branch for your changes.
+  2. If you are introducing a new api-version, create a new directory for that api-version and copy all the files from the previous version into the new directory. Make this the very first commit in your branch and then make your changes in subsequent commits.
```

Comment:
OpenAPI Hub can do this. Should we have them use that? It also updates the api-version for them. That said, the automatic stuff it does aren't always well-understood if people haven't used them.

Reply:
OpenAPI Hub can do this, but the tools require MS alias to login, and the introduction wiki is in Azure SDK internal wiki repo, which is only accessible by MS employee.   Proposed wording could be 

Microsoft employees can try out the experience at [OpenAPI Hub](https://aka.ms/openapihub) for [adding a new API version using OpenAPI Hub](https://dev.azure.com/azure-sdk/internal/_wiki/wikis/internal.wiki/208/OpenAPI-Hub-Adding-new-API-version).

Reply:
Added.